Fix arch used Apt.check with host_pacakge=False, when cross-compiling

sveinuah opened the pull request:
Changelog: Bugfix: Fixes an issue where the Apt packages for the build arch would be reported missing, in cross-compiling scenarios, even though they are installed.
Docs: Omit

Fixes #19073

sveinuah committed:
Add Apt.check tests with host_package=False

sveinuah committed:
Fix Apt dpkg query arch when host_package=False

Apt.check(<pkg>, host_package=False) added the host arch,
instead of the build arch to the dpkg query, if used in
a recipe during cross-building conan install invocations.
This fix ensures that the `settings_build.arch` is used
to make the query in this context.

Issue: https://github.com/conan-io/conan/issues/19073

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/conan-io/conan?pullRequest=19074) <br/>All committers have signed the CLA.

czoido commented on conan/tools/system/package_manager.py:
```diff
@@ -227,7 +227,13 @@ def _check(self, packages, host_package=True):
+        if host_package:
+            check_arch = self._arch or self._conanfile.settings_build.get_safe('arch')
+        else:
+            check_arch = self._conanfile.settings_build.get_safe('arch')
```
Hi @sveinuah,
Thanks a lot for the contribution. I think we could simplify the code to this?

```suggestion
        check_arch = self._arch if host_package else self._conanfile.settings_build.get_safe('arch')
```
Because when `host_package=True`, we should use the host architecture (`self._arch`) and not fallback to `settings_build`.

sveinuah replied:
That was my original thought as well, but the `test_tools_check` tests failed when I tried. I kept the old implementation in the `host_package` branch, in case there was a specific reason for it.


```
test/integration/tools/system/package_manager_test.py:449: AssertionError
____________ test_tools_check_with_version[Apt-dpkg-query -W -f='${Architecture} ${Version}\\n' package | grep -qEx '(amd64|all) 0.1']
...

E       assert "dpkg-query -...one|all) 0.1'" == "dpkg-query -...d64|all) 0.1'"
E         Skipping 60 identical leading characters in diff, use -v to show
E         - ep -qEx '(amd64|all) 0.1'
E         ?           ^^^^^
E         + ep -qEx '(None|all) 0.1'
E         ?  
```

sveinuah replied:
Ah. It may just have been a missing setting in the test. Adding the patch below, makes the tests pass.
Should I add this to the PR, and apply your suggestion?

Adding
```patch
diff --git a/test/integration/tools/system/package_manager_test.py b/test/integration/tools/system/package_manager_test.py
index e6b40c51a..ad9360731 100644
--- a/test/integration/tools/system/package_manager_test.py
+++ b/test/integration/tools/system/package_manager_test.py
@@ -439,7 +439,7 @@ def test_tools_install_archless_with_version(tool_class, result):
 ])
 def test_tools_check(tool_class, result):
     conanfile = ConanFileMock()
-    conanfile.settings = Settings()
+    conanfile.settings = MockSettings({"arch": "x86_64"})
     conanfile.conf.define("tools.system.package_manager:tool", tool_class.tool_name)
     with mock.patch('conan.ConanFile.context', new_callable=PropertyMock) as context_mock:
         context_mock.return_value = "host"
@@ -463,7 +463,7 @@ def test_tools_check(tool_class, result):
 ])
 def test_tools_check_with_version(tool_class, result):
     conanfile = ConanFileMock()
-    conanfile.settings = Settings()
+    conanfile.settings = MockSettings({"arch": "x86_64"})
     conanfile.conf.define("tools.system.package_manager:tool", tool_class.tool_name)
     with mock.patch('conan.ConanFile.context', new_callable=PropertyMock) as context_mock:
         context_mock.return_value = "host"
 ```

czoido replied:
Yes, I think that the test should have the `MockSettings({"arch": "x86_64"}` defined, that's the reason the test was failing.

sveinuah committed:
Add missing arch setting in test_tools_check

The mocked settings was missing an `arch`,
causing `_SystemPackageManagerTool._arch` to
be unset in the tests, leading to erroneous
test failures.

sveinuah commented:
Updated PR with the `MockSettings`, and your suggestion @czoido. Should be good to go from my side now :)

czoido commented on test/integration/tools/system/package_manager_test.py:
```diff
@@ -463,11 +463,51 @@ def test_tools_check(tool_class, result):
+        tool = tool_class(conanfile)
+        tool.check(["package=0.1"], host_package=False)
+
+    assert tool._conanfile.command == result
```
```suggestion
    assert tool._conanfile.command == result

```

czoido committed:
Apply suggestion from @czoido
